frontend/src/utils/content-safety.ts

```diff
@@ -2,22 +2,43 @@
  * Contains utilities related to content safety.
  */
 
-import type { Media } from "~/types/media"
 import { hash, rand as prng } from "~/utils/prng"
 import { log } from "~/utils/console"
+import {
+  USER_REPORTED,
+  PROVIDER_SUPPLIED,
+  TEXT_FILTERED,
+  Sensitivity,
+} from "~/constants/content-safety"
 
 /**
- * Marks the given item as mature based on a random number seeded with the
- * item's UUID v4 identifier. The `frac` param controls the probability of an
- * item being marked as mature.
+ * Get an array of randomly selected sensitive content flags for an item with
+ * the given UUID v4 identifier. The `frac` param controls the probability of an
+ * item having sensitivity flags.
  *
- * @param item - the item to mark as mature
- * @param frac - the fraction of items to mark as mature
+ * @param id - the ID of the item for which to calculate the flags
+ * @param frac - the fraction of items to probabilistically flag
+ * @returns an array of strings representing the mature flags
  */
-export const markFakeSensitive = (item: Media, frac = 0.5) => {
-  const random = prng(hash(item.id))()
-  if (random < frac) {
-    item.mature = true
-    log("Fake mature", item.frontendMediaType, item.id)
+export const getFakeSensitivities = (id: string, frac = 0.5): Sensitivity[] => {
+  const random = prng(hash(id))()
+
+  if (random > frac) {
+    return []
+  }
+
+  const sensitivityMask = Math.floor((random * 7) / frac) + 1
+  const sensitivity: Sensitivity[] = []
+  if ((sensitivityMask & 4) !== 0) {
+    sensitivity.push(USER_REPORTED)
   }
+  if ((sensitivityMask & 2) !== 0) {
+    sensitivity.push(PROVIDER_SUPPLIED)
+  }
```

Comment:
Technically provider supplied doesn't exist in our production data right now, so this can be safely ignored, unless we plan on implementing copy variations for it now even though it won't be used.

Reply:
It's good to mock a little ahead imo.